validation.md

```diff
@@ -0,0 +1,152 @@
+# Improving Validation
+
+The goal of this change is to improve the validation in Promitor to make it easier
+for users to find problems with their configuration, and to make it easier to add
+validation into the deserializers.
+
+## Must Haves
+
+- A consistent output format that explains exactly what the problem is, and ideally
+  where it is.
+- We must indicate when unknown field names are specified so we can highlight typos.
+
+## Nice to Haves
+
+- A schema for the configuration files so we have a single source of truth for the
```

Comment:
Do we need a schema or is the codebase the schema?

Reply:
I don't think we need a schema - the only real advantage I can think of would be to provide auto complete in vs code, but I don't really think it's worth it given there's no great support for yaml auto complete without using extensions.

It made sense to me when I started to write this up, but not so much now.

Reply:
I think it makes sense but more a question of bandwidth so I'd rather not do it now, but an issue for it longer term can't hurt